src/TableFunctions/TableFunctionS3.h

```diff
@@ -37,8 +36,10 @@ class TableFunctionS3 : public ITableFunction
         return {"_path", "_file"};
     }
 
+    template <bool get_format_from_file = true>
+    static void parseArgumentsImpl(const String & error_message, ASTs & args, ContextPtr context, StorageS3::Configuration & configuration);
```

Comment:
why not make `get_format_from_file` as just an argument? this template adds extra complexity in code for no reason, this is usually used for performance, but in our case it will not make any difference.